src/mlpack/tests/CMakeLists.txt

```diff
@@ -170,6 +168,12 @@ add_executable(mlpack_test
   main_tests/test_helper.hpp
 )
 
+add_executable(mlpack_catch_test
+  main.cpp
```

Comment:
Can you add `serialization.cpp` and `serialization.hpp` here, serialization.hpp depends on `serialization.cpp`.

Reply:
Done, but they have boost testing, should I change them also 
But even if I change them I have to still link against boost since it has boost serialisation `BOOST_SERIALIZATION_NVP`

Reply:
For the sake of the prototype, I would just suggest copying `serialization.hpp` and `serialization.cpp` to `serialization_catch.hpp` and `serialization_catch.cpp`, then adapt those `_catch` versions to use Catch and add those to the `mlpack_catch_test` target.  If it works, as we adapt the rest of the codebase we can just replace `serialization.hpp` and `serialization.cpp` with those. :)